### PR TITLE
wallet2 API: added change_password

### PR DESCRIPTION
**Pull request by stoffu**

### CODE REVIEW

File: src/wallet/api/wallet.cpp [hunk 1]
```diff
@@ -690,6 +690,20 @@ bool WalletImpl::store(const std::string &path)
...
+    try
+    {
+        m_wallet->rewrite(m_wallet->path(), new_password);
+        m_wallet->store();
+    } catch (const std::exception &e) {
+        LOG_ERROR("Error while changing password: " << e.what());
+        m_status = Status_Error;
+        m_errorString = e.what();
+    }
+    return m_status == Status_Ok;
```
Comment (moneromooo-monero) [inline src/wallet/api/wallet.cpp]: m_status is not set on the sucess path.

Comment (moneromooo-monero) [reply src/wallet/api/wallet.cpp]: In fact, that might be on purpose to not lose a previous failure ? If that is the case, I'd return at start I think.

Comment (stoffu) [reply src/wallet/api/wallet.cpp]: Oops! Fixed

Comment (stoffu) [reply src/wallet/api/wallet.cpp]: Hmm, now I' confused which convention is the right one, e.g. `setPassword` vs `daemonBlockChainHeight`.

Comment (moneromooo-monero) [reply src/wallet/api/wallet.cpp]: Hmm. I just looked and I don't get the difference between setPassword and change_password. The new one doesn't save the password, which seems odd. Can you explain the intent for the new function and differences ?

**Comment by stoffu** [comment]
@moneromooo-monero Sorry, this was my sloppy job not carefully reading the existing code. `setPassword` just does the job. Closing.

**Pull request closed without merging**